File: 14. BST 6.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None
class BST:
    def __init__(self):
        self.root = None
    def insert(self,data):
        if self.root is None:
            self.root= Node(data)
        else:
            self.insertRecursion(self.root,data)
    def insertRecursion(self,node,data):
        if data<node.data:
            if node.left is None:
                node.left=Node(data)
            else:
                self.insertRecursion(node.left,data)
        else:
            if node.right is None:
                node.right=Node(data)
            else:
                self.insertRecursion(node.right,data)

    def leftView(self):
        q = [self.root]
        result = []
        while q:
            level_size = len(q)
            for i in range(level_size):
                node = q.pop(0)
                if i == 0:
                    result.append((node.data))
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
        print(*result)

n = int(input())
bst = BST()
for i in range(n):
    keyword = input()
    bst.insert(keyword)
bst.leftView()

File: 14. BST 6.java
import java.util.*;

class Node {
    String data;
    Node left;
    Node right;

    Node(String data) {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}
class BST {
    Node root;
    public void insert(String data) {
        if (root == null) {
            root = new Node(data);
        } else {
            insertRecursion(root, data);
        }
    }
    public void insertRecursion(Node node, String data) {
        if (data.compareTo(node.data) < 0) 
        {
            if (node.left == null) {
                node.left = new Node(data);
            } else {
                insertRecursion(node.left, data);
            }
        } 
        else 
        {
            if (node.right == null) {
                node.right = new Node(data);
            } else {
                insertRecursion(node.right, data);
            }
        }
    }
    public List<String> leftView() 
    {
        List<String> result = new ArrayList<>();
        if (root == null) return result;

        Queue<Node> q = new LinkedList<>();
        q.add(root);

        while (!q.isEmpty()) {
            int levelSize = q.size();

            for (int i = 0; i < levelSize; i++) {
                Node node = q.poll();

                if (i == 0) {
                    result.add(node.data);
                }

                if (node.left != null) q.add(node.left);
                if (node.right != null) q.add(node.right);
            }
        }
        return result;
    }
}

public class Main {
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();

        BST bst = new BST();

        for (int i = 0; i < n; i++) 
        {
            String keyword = sc.nextLine();
            bst.insert(keyword);
        }

        List<String> leftViewResult = bst.leftView();
        System.out.println(String.join(" ", leftViewResult));
    }
}
